docs/adr/0024-upgrade-templating-toolset-of-auth-server-emails.md

```diff
@@ -0,0 +1,237 @@
+
+## Decisions and Considered Options
+
+### Templating
+
+- Option A - Continue to use Mustache
+- Option B - Use React server-side to generate static pages
+- Option C - Use EJS
+
+### CSS styles
+
+- Option A - Continue using ad-hoc inline styles
+- Option B - Use TailwindCSS
+- Option C - Use CSS-in-JS
+- Option D - Use a CSS pre-processor like SASS
+
+### Email previewing
+
+- Option A - Continue to use the `write-emails` command and maildev
+- Option B - Use Storybook
+- Option C - Any other client that could be used to view emails? [todo]
+
+--
+
+Furthermore, there are a few **other decisions** worth noting that won’t necessarily have pros/cons lists:
+
+- How to handle the plaintext `.txt` files
+- Transitioning to Fluent over GetText for localization
+- Plans around integration involving feature flagging and the QA process
+
+## Decision Outcome
+
+### Templating
+
+Chosen option: "Option B - Use React server-side to generate static pages", because:
+
+- The team has experience with (and likes) React
+- We're already using it in other parts of the ecosystem, i.e. the payments server
+- React has been tried and tested to work well with Storybook
+- React can be integrated with mjml to generate responsive emails and there is already a great [library](https://github.com/wix-incubator/mjml-react) which can help us with that purpose.
+- Components across the templates can be reused if written in React
+
+A note about tests: Since we're going with the React-based solution for our templating stack, it may make sense to use Jest to test the templates. However, we don't want a ton of time spent in test land. At the very least, we can take what we currently have and transfer/copy it, and look them over to see if anything can be improved without a high level of effort.
+
+### MJML
+
+As part of modernizing the templates, we aim to replace the framework in which our templates are currently built in so as to include more logic in them while development. Currently, we've decided on using [MJML](https://mjml.io/) as our potential markup along with a templating framework like React, Mustache, or EJS. However, there needs to be more discussion around it so the following list of pros and cons could be pondered upon to decide whether or not we should use MJML as part of our templating stack.
+
+- Pros
```

Comment:
Ohh cool I'll move it down but then should I also include anything about MJML under Considered Options?
I'm thinking to just have a heading there and then detail about the pros and cons in subsequent section. Maybe something like this: 
```
## Decisions and Considered Options

### MJML

### Templating

- Option A - Continue to use Mustache
- Option B - Use React server-side to generate static pages
- Option C - Use EJS
```

Reply:
This seems good to me, or, sticking it as a subheader under "Templating" since it is related:

## Decisions and Considered Options

### Templating

- Option A - Continue to use Mustache
- Option B - Use React server-side to generate static pages
- Option C - Use EJS


#### MJML

We also evaluated whether or not to include MJML in the stack.

### CSS styles